### PR TITLE
Pair get info method

### PR DESCRIPTION
**Pull request by shivamgarglab49**

### CODE REVIEW

File: contracts/Pair.sol [hunk 1]
```diff
@@ -302,6 +298,21 @@ contract Pair is IPair, Initializable {
...
 
+    function getPairInfo()
+        public
+        view
+        returns (Pair memory _pair, Amount memory _amount)
+    {
+        _pair = pair;
+        _amount = Amount(
+            getSpotPrice(pair.tokenA.tokenAddress),
+            getSpotPrice(pair.tokenB.tokenAddress),
```
Comment (amandeeprehal-lab49) [inline contracts/Pair.sol:309]: WE can add lpTokenAddress as well.

Comment (amandeeprehal-lab49) [reply contracts/Pair.sol:309]: Plus getSlippage

Comment (amandeeprehal-lab49) [reply contracts/Pair.sol:309]: getContractAddress ?

File: contracts/Pair.sol [hunk 2]
```diff
@@ -310,8 +321,10 @@ contract Pair is IPair, Initializable {
         int256 otherTokenQty = token == pair.tokenA.tokenAddress
             ? pair.tokenB.tokenQty
             : pair.tokenA.tokenQty;
-        int256 value = (spotTokenQty * precision) / otherTokenQty;
-
+        int256 value;
+        if (otherTokenQty > 0) {
+            value = (spotTokenQty * precision) / otherTokenQty;
```
Comment (amandeeprehal-lab49) [inline contracts/Pair.sol:326]: otherTokenQty  == 0 then it will return 0 value which is wrong info, add a require check and let it fail

Comment (shivamgarglab49) [reply contracts/Pair.sol:326]: Actually UI can call getPairInfo() without adding the liquidity to read other values like 'getPrecisionValue','getFeePrecision',fee etc.

if we need to add revert condition then UI has to sure to call it only when pair having some quantities.

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**